src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/MessagingProvider.cs

```diff
@@ -93,6 +95,49 @@ public virtual MessageSender CreateMessageSender(string entityPath, string conne
             return GetOrAddMessageSender(entityPath, connectionString);
         }
 
+        /// <summary>
+        /// Creates a <see cref="ClientEntity"/> for the specified ServiceBus entity. It's used for sessions scenarios.
+        /// </summary>
+        /// <remarks>
+        /// You can override this method to customize the <see cref="ClientEntity"/>.
+        /// </remarks>
+        /// <param name="entityPath">The ServiceBus entity to create a <see cref="MessageSender"/> for.</param>
+        /// <param name="connectionString">The ServiceBus connection string.</param>
+        /// <returns></returns>
+        public virtual ClientEntity CreateClientEntity(string entityPath, string connectionString)
```

Comment:
consider adding some tests to the existing MessagingProviderTests for the new APIs